### PR TITLE
Performance and optimization vignette

### PR DESCRIPTION
**Pull request by yjunechoe**
@trangdata let's work on this branch for #274!

### CODE REVIEW

**Comment by trangdata** [comment]
@yjunechoe  I added the `oa_generate()` section and everything looks good! (Sorry for the sentence breaks... I used Visual mode — should have done it in two commits so it's easier for you to see what I changed.)

**Comment by trangdata** [comment]
Hmm oa_ngrams is failing because the endpoint is [currently not in service](https://docs.openalex.org/api-entities/works/get-n-grams#api-endpoint):

> https://docs.openalex.org/api-entities/works/get-n-grams#api-endpoint

**Review by yjunechoe — COMMENTED** [review]
Just leaving 1 thought - up to you to see if it fits!

File: vignettes/articles/performance-optimization.Rmd [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+This way, you can process records in batches of, say, 1000 records, and write them out to disk as you go along.[^2]
+
+[^2]: Quick note: For the case without `group_by`, setting `verbose = TRUE` when you call `oar()` for the first time will give the total number of records in the message.
+    With `group_by`, unfortunately we don't have a way to know the total until we query until exhausted.
+    Also, you will need to have the R package [**coro**](https://github.com/r-lib/coro) installed to use `oa_generate()` (`install.packages("coro")`).
+
+In the example below, we show how `oa_generate()` works when we want to find all the works that cite [W1160808132](https://openalex.org/works/W1160808132).
+
+```{r}
+query_url <- "https://api.openalex.org/works?filter=cites%3AW1160808132"
```
[the remark this answers is not in the document]
Comment (trangdata) [reply vignettes/articles/performance-optimization.Rmd:143]: Great point, June! Incorporated in f88a2aa.

**Review by trangdata — APPROVED** [review]

**Pull request merged**